### PR TITLE
Allow pod sec context to be overridden

### PR DESCRIPTION
**Pull request by skonto**
Address partially #505, at least on Openshift.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -50,23 +50,24 @@ import (
...
-	webhookCertDir      = flag.String("webhook-cert-dir", "/etc/webhook-certs", "The directory where x509 certificate and key files are stored.")
-	webhookSvcNamespace = flag.String("webhook-svc-namespace", "spark-operator", "The namespace of the Service for the webhook server.")
-	webhookSvcName      = flag.String("webhook-svc-name", "spark-webhook", "The name of the Service for the webhook server.")
-	webhookPort         = flag.Int("webhook-port", 8080, "Service port of the webhook server.")
-	enableMetrics       = flag.Bool("enable-metrics", false, "Whether to enable the metrics endpoint.")
-	metricsPort         = flag.String("metrics-port", "10254", "Port for the metrics endpoint.")
-	metricsEndpoint     = flag.String("metrics-endpoint", "/metrics", "Metrics endpoint.")
-	metricsPrefix       = flag.String("metrics-prefix", "", "Prefix for the metrics.")
-	ingressUrlFormat    = flag.String("ingress-url-format", "", "Ingress URL format.")
+	master                = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
```
Comment (skonto) [inline main.go]: go fmt ./... reformatted this.

File: main.go [hunk 2]
```diff
@@ -50,23 +50,24 @@ import (
...
+	webhookConfigName     = flag.String("webhook-config-name", "spark-webhook-config", "The name of the MutatingWebhookConfiguration object to create.")
+	webhookCertDir        = flag.String("webhook-cert-dir", "/etc/webhook-certs", "The directory where x509 certificate and key files are stored.")
+	webhookSvcNamespace   = flag.String("webhook-svc-namespace", "spark-operator", "The namespace of the Service for the webhook server.")
+	webhookSvcName        = flag.String("webhook-svc-name", "spark-webhook", "The name of the Service for the webhook server.")
+	webhookPort           = flag.Int("webhook-port", 8080, "Service port of the webhook server.")
+	enableMetrics         = flag.Bool("enable-metrics", false, "Whether to enable the metrics endpoint.")
+	metricsPort           = flag.String("metrics-port", "10254", "Port for the metrics endpoint.")
+	metricsEndpoint       = flag.String("metrics-endpoint", "/metrics", "Metrics endpoint.")
+	metricsPrefix         = flag.String("metrics-prefix", "", "Prefix for the metrics.")
+	overridePodSecContext = flag.Bool("override-pod-sec-context", false, "Whether to override existing pod security context with patching.")
```
Comment (liyinan926) [inline main.go]: I don't think we need a flag to control this. Always overriding as long as users explicitly specify a `PodSecurityContext` in their `SparkApplication`s seems fine to me.

Comment (skonto) [reply main.go]: Ok sure I had it this way but was not sure if people want more control.

**Comment by skonto** [comment]
@liyinan926 ready to merge. :)

**Comment by Reamer** [comment]
If you on Openshift. You must enable ValidatingAdmissionWebhook and MutatingAdmissionWebhook.
Howto: https://github.com/openshift/openshift-ansible/issues/7983#issuecomment-382505676

**Comment by anandbabus01** [comment]
I'm still facing problem. add below values along with memory and core but sill driver and executor runs in root user.
```
spec:
  driver:
    securityContext:
      runAsUser: 1000    
  executor:
    securityContext:
      runAsUser: 1000
```
web-hook service is running in spark-operator name space. now using below command to deploy spark app.
kubectl apply -f spark/template/spark-application.yaml.

any help appreciated.

**Pull request merged**